runtime/jcl/common/getstacktrace.c

```diff
@@ -35,23 +35,33 @@
 
 
 j9object_t
-getStackTraceForThread(J9VMThread *currentThread, J9VMThread *targetThread, UDATA skipCount)
+getStackTraceForThread(J9VMThread *currentThread, J9VMThread *targetThread, UDATA skipCount, j9object_t threadObject)
 {
 	J9JavaVM * vm = currentThread->javaVM;
 	J9InternalVMFunctions * vmfns = vm->internalVMFunctions;
 	j9object_t throwable = NULL;
 	J9StackWalkState walkState;
-	UDATA rc;
+	UDATA rc = J9_STACKWALK_RC_NONE;
 
 	/* Halt the target thread */
 	vmfns->haltThreadForInspection(currentThread, targetThread);
 
 	/* walk stack and cache PCs */
-	walkState.walkThread = targetThread;
 	walkState.flags = J9_STACKWALK_CACHE_PCS | J9_STACKWALK_WALK_TRANSLATE_PC | J9_STACKWALK_SKIP_INLINES | J9_STACKWALK_INCLUDE_NATIVES | J9_STACKWALK_VISIBLE_ONLY;
-	walkState.skipCount = skipCount;
-	rc = vm->walkStackFrames(currentThread, &walkState);
-
+#if JAVA_SPEC_VERSION >= 19
+	if ((NULL != targetThread->currentContinuation) && (threadObject != targetThread->threadObject)) {
```

Comment:
The second check is still required to determine if the caller wants the carrier stack or virtual stack

Reply:
Ok. It is unclear what is the input `threadObject`, and the difference between current and target `threadObject`. Probably, because the function description is missing for `getStackTraceForThread`. Also, whose stack trace is being retrieved?

Reply:
`(threadObject != targetThread->threadObject)`: How do we guarantee that `threadObject` is a carrier thread and not some other thread?

Reply:
when a virtual thread is mounted to a carrier, they share the same J9VMThread. So `Thread.getStackTraceImpl` can be called on either virtual/carrier object and will pass the same `targetThread` to `getStackTraceForThread`. by checking if the `threadObject` (ie. receiverObject `Thread.getStackTraceImpl` is called on) matches the current thread object, we ensure the virtual stack is not returned as carrier stack.

>(threadObject != targetThread->threadObject): How do we guarantee that threadObject is a carrier thread and not some other thread?

the caller guarantees this, targetThread is retrieved from the threadObject being passed in.

Reply:
We should add a function description adn comments to document the above behaviour, especially the caller/callee dependence.

Reply:
this function is only used by `getStackTraceImpl` so it is more of an abstraction than caller guarantee.
But yes, we should add a doc so future updates would be easier